the-collab-lab/tcl-79-smart-shopping-list
# Relocate Add Item Functionality to Modal on List Page

## Description

The add item functionality has been relocated to the list page, where it is now accessed via a modal triggered by clicking the "+" next to the search bar. This modal includes an option for specifying item quantity.

## Related Issue

closes #33 

## Acceptance Criteria

- [x] Move the add item form to list page.
- [x] Prevent adding only numbers as item names, currently the form allows numbers to be item names and it will submit it as empty item list name.
- [x] Add item quantity. 
- [x] Create a modal for add item to be displayed when the add button is clicked.
- [x] Clear input button should be removed following the design. 


## Type of Changes

Enhancement

## Updates

### Before

![Screen Shot 2024-10-03 at 1 51 13 PM](https://github.com/user-attachments/assets/4306fcc3-277b-449e-ae2e-a3ea908dc6ff)
![Screen Shot 2024-10-03 at 2 05 54 PM](https://github.com/user-attachments/assets/2d4733be-6df4-4fa0-94df-169171f2d62c)



### After

![Screen Shot 2024-10-03 at 2 05 18 PM](https://github.com/user-attachments/assets/4e87d600-d008-49a9-9042-f7c8e12f2560)
![Screen Shot 2024-10-03 at 2 05 25 PM](https://github.com/user-attachments/assets/3333887a-f50e-4255-80d0-647bedff04ec)




## Testing Steps / QA Criteria

- Do a `git pull` and `git checkout hm-addItemForm-relocation-and-modifications`
- Open the homepage by running `npm start`.
- Navigate to the `list` page.
- The add item functionality should now be in a modal triggered by the "+" button next to the search bars with an option to specify item quantity.
